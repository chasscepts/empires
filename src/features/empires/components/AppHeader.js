import { Link } from 'react-router-dom';

const styles = {
  header: {
    display: 'flex',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
};

export default function AppHeader() {
  return (
    <header style={styles.header}>
      <h1>Age Of Empires</h1>
      <nav>
        <Link to="/">Home</Link>
      </nav>
    </header>
  );
}
