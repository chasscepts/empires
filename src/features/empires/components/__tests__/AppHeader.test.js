import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import RouterWrapper from '../test_helpers/RouterWrapper';
import AppHeader from '../AppHeader';

it('matches snapshot', () => {
  const tree = renderer
    .create(<RouterWrapper component={AppHeader} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

describe('AppHeader', () => {
  beforeEach(() => {
    render(<RouterWrapper component={AppHeader} />);
  });

  it('displays the heading', () => {
    expect(screen.getByText('Age Of Empires II')).toBeInTheDocument();
  });

  it('displays the home link', () => {
    expect(screen.getByRole('link')).toBeInTheDocument();
  });
});
