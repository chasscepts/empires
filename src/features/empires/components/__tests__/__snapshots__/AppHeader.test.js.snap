// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<header
  className="header"
>
  <h1>
    Age Of Empires II
  </h1>
  <nav>
    <a
      className="link"
      href="/"
      onClick={[Function]}
      role="link"
    >
      🏠
    </a>
  </nav>
</header>
`;
